File: html-report/src/components/HomePage.js
import React, {Component} from 'react';
import convertTime from './../utils/convertTime';
import paths from './../utils/paths';

export default class HomePage extends Component {

    componentDidMount() {
        document.title = window.mainData.title;
    }

    render() {
        return (
            <div className="content margin-top-20">
                <div className="title-common">Pools</div>

                {window.mainData.pools.map((pool) => {
                        return (
                            <div key={pool.id} className="suite-item card">
                                <a href={paths.fromIndexToPool(pool.id)} className="title-common with-arrow">
                                    Pool {pool.id}
                                </a>
                                <div className="row full margin-bottom-20 bounded">
                                    <div className="card-info">
                                        <div className="text-sub-title-light">Passed</div>
                                        <div className="card-info__content status-passed">{pool.passed_count}</div>
                                    </div>
                                    <div className="card-info">
                                        <div className="text-sub-title-light">Failed</div>
                                        <div className="card-info__content status-failed">{pool.failed_count}</div>
                                    </div>
                                    <div className="card-info">
                                        <div className="text-sub-title-light">Ignored</div>
                                        <div className="card-info__content status-ignored">{pool.ignored_count}</div>
                                    </div>
                                    <div className="card-info">
                                        <div className="text-sub-title-light">Duration</div>
                                        <div className="card-info__content">{convertTime(pool.duration_millis)}</div>
                                    </div>
                                    <div className="card-info">
                                        <div className="text-sub-title-light">Devices</div>
                                        <div className="card-info__content">{pool.devices.length}</div>
                                    </div>
                                </div>
                            </div>
                        )
                    }
                )}

                <div className="title-common">Timeline</div>

                <div className="suite-item card">
                    <iframe src="timeline/index.html" style={{ width: "100%", height: "137" }}></iframe>
                </div>
            </div>
        );
    }
}

File: report/marathon/src/main/kotlin/marathon/component/HomePage.kt
package marathon.component

import marathon.model.PoolSummary
import marathon.model.Run
import marathon.model.Status
import react.FC
import react.Props
import react.dom.html.ReactHTML.a
import react.dom.html.ReactHTML.div
import react.key

external interface HomeProps : Props {
    var run: Run
}

val HomePage = FC<HomeProps> {
    div {
        className = "content margin-top-20"

        div {
            className = "title-common"
            +"Pools"
        }

        for (pool in it.run.pools) {
            Pool {
                this.pool = pool
            }
        }
    }
}

external interface PoolProps : Props {
    var pool: PoolSummary
}

val Pool = FC<PoolProps> { props ->
    div {
        key = props.pool.id
        className = "suite-item card"

        a {
            href = "somewhere"
            className = "title-common with-arrow"
            +"Pool ${props.pool.id}"
        }

        div {
            className = "row full margin-bottom-20 bounded"

            Card {
                title = "Passed"
                status = Status.Passed
                value = props.pool.passed_count.toString()
            }

            Card {
                title = "Failed"
                status = Status.Failed
                value = props.pool.failed_count.toString()
            }

            Card {
                title = "Ignored"
                status = Status.Ignored
                value = props.pool.ignored_count.toString()
            }

            Card {
                title = "Duration"
                value = props.pool.duration_millis.toString()
            }

            Card {
                title = "Devices"
                value = props.pool.devices.size.toString()
            }
        }
    }
}

external interface CardProps : Props {
    var title: String
    var status: Status?
    var value: String
}

val Card = FC<CardProps> { props ->
    div {
        className = "card-info"

        div {
            className = "text-sub-title-light"
            +props.title
        }

        div {
            className = "card-info__content" + when (props.status) {
                Status.Passed -> " status-passed"
                Status.Failed -> " status-failed"
                Status.Ignored -> " status-ignored"
                else -> ""
            }
            +props.value
        }
    }
}
